compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4810,7 +4810,7 @@ object Types {
     def hasLowerBound(using Context): Boolean = !currentEntry.loBound.isExactlyNothing
 
     /** For uninstantiated type variables: Is the upper bound different from Any? */
-    def hasUpperBound(using Context): Boolean = !currentEntry.hiBound.finalResultType.isExactlyAny
+    def hasUpperBound(using Context): Boolean = !currentEntry.hiBound.finalResultType.isRef(defn.AnyClass) // so also `Any[..]`
```

Comment:
This is weird: Any is not kind-polymorphic, it's not supposed to be applicable to arguments. I've opened a PR which I believe fixes the root issue: https://github.com/lampepfl/dotty/pull/17025